### ISSUE TITLE
udftools build fail

### ISSUE THREAD

**Issue by fengjiannan2010** [pali/udftools#19]
os centos 6.5
[root@oOcean udftools]# ./autogen.sh
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, `config'.
libtoolize: copying file `config/ltmain.sh'
libtoolize: Consider adding `AC_CONFIG_MACRO_DIR([m4])' to configure.ac and
libtoolize: rerunning libtoolize, to keep the correct libtool macros in-tree.
libtoolize: Consider adding `-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
configure.ac:11: installing `config/config.guess'
configure.ac:11: installing `config/config.sub'
configure.ac:6: installing `config/install-sh'
configure.ac:6: installing `config/missing'
cdrwtool/Makefile.am: installing `config/depcomp'

[root@oOcean udftools]# ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables...
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking dependency style of gcc... gcc3
checking for gcc option to accept ISO C99... -std=gnu99
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc -std=gnu99... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1966080
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for ar... ar
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc -std=gnu99 object... ok
checking how to run the C preprocessor... gcc -std=gnu99 -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc -std=gnu99 supports -fno-rtti -fno-exceptions... no
checking for gcc -std=gnu99 option to produce PIC... -fPIC -DPIC
checking if gcc -std=gnu99 PIC flag -fPIC -DPIC works... yes
checking if gcc -std=gnu99 static flag -static works... no
checking if gcc -std=gnu99 supports -c -o file.o... yes
checking if gcc -std=gnu99 supports -c -o file.o... (cached) yes
checking whether the gcc -std=gnu99 linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for readline in -lreadline... yes
./configure: line 11345: syntax error near unexpected token `fi'
./configure: line 11345: `fi'
[root@oOcean udftools]#

**Comment by pali** [comment]
Which version of udftools?

And can you use pre-generated script `configure` instead of generating a new via `autogen.sh`

**Comment by fengjiannan2010** [comment]
Hello:
      The version I built is master. Now I can build successfully using release version 2.0. Thank you.



fengjiannan2010@163.com
 
From: pali
Date: 2018-05-15 16:10
To: pali/udftools
CC: fengjiannan2010; Author
Subject: Re: [pali/udftools] udftools build fail (#19)
Which version of udftools?
And can you use pre-generated script configure instead of generating a new via autogen.sh

**Comment by fengjiannan2010** [comment]
Hello:
      The version I built is master. Now I can build successfully using release version 2.0. Thank you.

**Comment by pali** [comment]
Ok, can you bisect git commits between 2.0 and master to find which commit broke it?

**Comment by pali** [comment]
And can you provide your generated `configure` script which show that syntax error? Lines around 11345 should be enough.

**Comment by pali** [comment]
Relevant lines of `configure` script are:
```
11343 if test "x$ac_cv_lib_readline_readline" = x""yes; then
11344 
11345 fi
```

There is missing body of `if then .. fi` condition.

This is a bug in autoconf which was fixed in autoconf version 2.64. Also due to usage of PACKAGE_URL is autoconf 2.64 required. I will update it in AC_PREREQ.

On CentOS 6 is default version of autoconf just 2.63, therefore you hit above bug. But CentOS 6 has package `autoconf268` which contains autoconf 2.68, so use this one.
